src/main/java/com/aws/greengrass/util/GreengrassServiceClientFactory.java

```diff
@@ -87,12 +94,25 @@ private boolean validString(Node node, String key) {
         return node != null && node.childOf(key) && Utils.isNotEmpty(Coerce.toString(node));
     }
 
+    /**
+     * Retrieve configValidationError.
+     * Validate again if the device config has changed.
+     *
+     */
+    public String getConfigValidationError() {
+        if (deviceConfigChanged.compareAndSet(true, false)) {
+            validateConfiguration();
+        }
+        return configValidationError;
+    }
+
     /**
      * Initializes and returns GreengrassV2DataClient.
      *
      */
     public synchronized GreengrassV2DataClient getGreengrassV2DataClient() {
```

Comment:
since you are making changes here, can you update the JavaDoc to indicate that this can return `null` if there is a config validation error

Reply:
Will update.